src/pages/archives/how-to-configure-astropaper-theme.md
---
author: Sat Naing
pubDatetime: 2022-09-23T04:58:53Z
modDatetime: 2024-10-14T09:27:28.605Z
title: How to configure AstroPaper theme
slug: how-to-configure-astropaper-theme
featured: true
draft: false
tags:
  - configuration
  - docs
description: How you can make AstroPaper theme absolutely yours.
---

Yep AstroPaper is a highly customizable Astro blog theme. With AstroPaper, you can customize everything according to your personal taste. This article will explain how you can make some customizations easily in the config file.

## Table of contents

## Configuring SITE

The important configurations lies in `src/config.ts` file. Within that file, you'll see the `SITE` object where you can specify your website's main configurations.

During development, it's okay to leave `SITE.website` empty. But in production mode, you should specify your deployed url in `SITE.website` option since this will be used for canonical URL, social card URL etc.. which are important for SEO.

```js
// file: src/config.ts
export const SITE = {
  website: "https://astro-paper.pages.dev/",
  author: "Sat Naing",
  desc: "A minimal, responsive and SEO-friendly Astro blog theme.",
  title: "AstroPaper",
  ogImage: "astropaper-og.jpg",
  lightAndDarkMode: true,
  postPerPage: 3,
  scheduledPostMargin: 15 * 60 * 1000, // 15 minutes
  showArchives: true,
  editPost: {
    url: "https://github.com/satnaing/astro-paper/edit/main/src/content/blog",
    text: "Suggest Changes",
    appendFilePath: true,
  },
};
```

Here are SITE configuration options

| Options               | Description                                                                                                                                                                                                                                                                                                                                        |
| --------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `website`             | Your deployed website url                                                                                                                                                                                                                                                                                                                          |
| `author`              | Your name                                                                                                                                                                                                                                                                                                                                          |
| `desc`                | Your site description. Useful for SEO and social media sharing.                                                                                                                                                                                                                                                                                    |
| `title`               | Your site name                                                                                                                                                                                                                                                                                                                                     |
| `ogImage`             | Your default OG image for the site. Useful for social media sharing. OG images can be an external image url or they can be placed under `/public` directory.                                                                                                                                                                                       |
| `lightAndDarkMode`    | Enable or disable `light & dark mode` for the website. If disabled, primary color scheme will be used. This option is enabled by default.                                                                                                                                                                                                          |
| `postPerIndex`        | The number of posts to be displayed at the home page under `Recent` section.                                                                                                                                                                                                                                                                       |
| `postPerPage`         | You can specify how many posts will be displayed in each posts page. (eg: if you set SITE.postPerPage to 3, each page will only show 3 posts per page)                                                                                                                                                                                             |
| `scheduledPostMargin` | In Production mode, posts with a future `pubDatetime` will not be visible. However, if a post's `pubDatetime` is within the next 15 minutes, it will be visible. You can set `scheduledPostMargin` if you don't like the default 15 minutes margin.                                                                                                |
| `showArchives`        | Determines whether to display the `Archives` menu (positioned between the `About` and `Search` menus) and its corresponding page on the site. This option is set to `true` by default.                                                                                                                                                             |
| `editPost`            | This option allows users to suggest changes to a blog post by providing an edit link under blog post titles. This feature can be disabled by removing it from the `SITE` config. You can also set `appendFilePath` to `true` to automatically append the file path of the post to the url, directing users to the specific post they wish to edit. |

## Configuring locale

You can configure the default locale used for the build (e.g., date format in the post page), and for the rendering in browsers (e.g., date format in the search page)

```js
// file: src/config.ts
export const LOCALE = {
  lang: "en", // html lang code. Set this empty and default will be "en"
  langTag: ["en-EN"], // BCP 47 Language Tags. Set this empty [] to use the environment default
} as const;
```

`LOCALE.lang` will be used as HTML ISO Language code in `<html lang="en">`. If you don't specify this, default fallback will be set to `en`.
`LOCALE.langTag` is used as [datetime locale](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString#locales). For this, you can specify an array of locales for fallback languages. Leave `LOCALE.langTag` empty `[]` to use the environment default at _build-_ and _run-time_.

## Configuring logo or title

You can specify site's title or logo image in `src/config.ts` file.

![An arrow pointing at the website logo](https://res.cloudinary.com/noezectz/v1663911318/astro-paper/AstroPaper-logo-config_goff5l.png)

```js
// file: src/config.ts
export const LOGO_IMAGE = {
  enable: false,
  svg: true,
  width: 216,
  height: 46,
};
```

If you specify `LOGO_IMAGE.enable` => `false`, AstroPaper will automatically convert `SITE.title` to the main site text logo.

If you specify `LOGO_IMAGE.enable` => `true`, AstroPaper will use the logo image as the site's main logo.

You have to specify `logo.png` or `logo.svg` under `/public/assets` directory. Currently, only svg and png image file formats are supported. (**_Important!_** _logo name has to be logo.png or logo.svg)_

If your logo image is png file format, you have to set `LOGO_IMAGE.svg` => `false`.

It is recommended that you specify width and height of your logo image. You can do that by setting `LOGO_IMAGE.width` _and_ `LOGO_IMAGE.height`

## Configuring social links

You can configure your own social links along with its icons.

![An arrow pointing at social link icons](https://res.cloudinary.com/noezectz/v1663914759/astro-paper/astro-paper-socials_tkcjgq.png)

Currently 20 social icons are supported. (Github, LinkedIn, Facebook etc.)

You can specify and enable certain social links in hero section and footer. To do this, go to `/src/config.ts` and then you'll find `SOCIALS` array of object.

```js
// file: src/config.ts
export const SOCIALS: SocialObjects = [
  {
    name: "Github",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: ` ${SITE.title} on Github`,
    active: true,
  },
  {
    name: "Facebook",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Facebook`,
    active: true,
  },
  {
    name: "Instagram",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Instagram`,
    active: true,
  },
  ...
]
```

You have to set specific social link to `active: true` in order to appear your social links in hero and footer section. Then, you also have to specify your social link in `href` property.

For instance, if I want to make my Github appear, I'll make it like this.

```js
export const SOCIALS: SocialObjects = [
  {
    name: "Github",
    href: "https://github.com/satnaing", // update account link
    linkTitle: `${SITE.title} on Github`, // this text will appear on hover and VoiceOver
    active: true, // makre sure to set active to true
  }
  ...
]
```

Another thing to note is that you can specify the `linkTitle` in the object. This text will display when hovering on the social icon link. Besides, this will improve accessibility and SEO. AstroPaper provides default link title values; but you can replace them with your own texts.

For example,

```js
linkTitle: `${SITE.title} on Twitter`,
```

to

```js
linkTitle: `Follow ${SITE.title} on Twitter`;
```

## Conclusion

This is the brief specification of how you can customize this theme. You can customize more if you know some coding. For customizing styles, please read [this article](https://astro-paper.pages.dev/posts/customizing-astropaper-theme-color-schemes/). Thanks for reading.✌🏻


content/note/2023-12-05/configuration-facile-de-xdebug-en-cli.md
---
title: "Configuration Facile De Xdebug en Cli "
date: 2023-12-05T14:54:50Z
draft: false
tags: ["php","xdebug","docker", "phpstorm"]
slug: "configuration-facile-de-xdebug-en-cli"
image: "/home/app/content/note/2023-12-05/configuration-facile-de-xdebug-en-cli.md/ogimage.png"
---

J'ai pris un peu de temps pour creuser l'utilisation de xdebug avec les commandes en cli interactives et phpstorm.

<!--more-->

Pour le contexte, j'utilise [phpstorm](/tags/phpstorm) et j'ai un petit container [docker](/tags/docker) avec mon appli php. C'est une appli en cli (command line interface) interactive.
Mon intention c'est de faire du _step debugging_, c'est à dire mettre des _breakpoints_ pour voir déboger le script.

## C'est peut-être compliqué ...

La [doc officielle de xdebug](https://xdebug.org/docs/step_debug) et la [doc de phpstorm](https://www.jetbrains.com/help/phpstorm/configuring-xdebug.html) sont très touffues et on s'y perd. On s'y perd vraiment si on va chercher dans les [paramètres](https://xdebug.org/docs/all_settings) pourquoi ça ne va. Il y a de quoi se décourager si on est pressé de mettre son enviromment de developpement en place. Je sais que beaucoup vont préférer lancer de vulgaires die(), var_dump(),exceptions à la place. C'est pas très pro et c'est surtout bien moins efficace, en particulier si on doit chercher à quel moment ça part en sucette.

On a 3 façon de paramétrer xdebug. Dans un fichier d'initialisation de php (`99-xdebug.ini`, classiquement), dans les paramètres de la ligne commande `php -dxdebug_client_host=172.18.0.1 ... monscript.php` et dans les variables d'environnement `XDEBUG_CONFIG="client_host=192.168.42.34 log=/tmp/xdebug.log" php monscript.php`. En thèorie c'est pratique, en pratique ajoute inutilement de l'information. Ça peut être d'autant plus perturbant quand on sait qu'on peut mixer les 3 approches.
Dans [le zen de python](https://fr.wikipedia.org/wiki/Zen_de_Python) on peut lire 
> Sache qu'il ne devrait y avoir qu'une et une seule façon de procéder

C'est à mon avis une règle très importante, dans le doute, je m'y réfère souvent, j'imagine que c'est un principe que ne renierait pas [Oncle Bob](https://www.cultura.com/p-coder-proprement-9782326002272.html). 



## Mais en fait, c'est pas très compliqué

Il y a deux choses à faire, lancer le script et configurer phpstorm pour qu'il écoute les infos.

Tout d'abord, dans phpstorm, ça fonctionne _out of the box_, on peut lancer une commande php en mode debugage et tout est ok. (Run > debug).  
Ça pourrait me suffire si mon script n'était pas une commande interactive. Mais pour un script interactif, on est bloqué (ou alors il faut m'expliquer). Et puis savoir ne pas être dépendant de son ide est une bonne chose.

### La configuration de phpstorm

Quand phpstorm lance lui même la commande, il met tout ce qu'il faut en variable d'environnement et il sait quel fichier est exécuté.  
Quand ça n'est pas phpstorm qui lance la commande, il ne va faire que recevoir les infos de xdebug, c'est à nous de définir les variables d'environnement, mais il faut aussi indiquer un _mapping_ entre le fichier executé et nos fichiers locaux.  
Pour ça il faut définin un serveur dans phpstorm (pas un serveur d'outils, pas un serveur de déploiement (il y a beaucoup de choses dans phpstorm)). 

Pour créér un serveur dans phpstorm : _Configuration_ (CTRL+S) > PHP > Servers : _ajouter un serveur_, cocher `use path mapping` et indiquer la localisation correspondant à la racine de votre projet.  
Ce serveur il aura un petit nom, c'est ce nom qu'il faut utiliser dans la variable d'environnement `PHP_IDE_CONFIG`, par exemple : `PHP_IDE_CONFIG="serverName=myrepl"` (myrepl est le nom de mon _server_ créé à l'étape précédente dans phpstorm).

XDEBUG_MODE=debug XDEBUG_SESSION=1 XDEBUG_CONFIG="client_host=172.17.0.1 client_port=9003" PHP_IDE_CONFIG="serverName=myrepl" php /app/hello.php
# 

### Lancement du script

J'ai choisi d'utiliser les variables d'environnement pour la configuration de xdebug. Je reviendrai peut-être sur ce choix à l'avenir, pour le moment ça fait le taf.  
Dans le terminal de mon contenaire, je lance `XDEBUG_MODE=debug XDEBUG_SESSION=1 XDEBUG_CONFIG="client_host=172.17.0.1 client_port=9003" PHP_IDE_CONFIG="serverName=myrepl" php /app/hello.php` et c'est tout bon, phpstorm s'arretera sur le premier breakpoint défini. Je vais pouvoir regarder l'état des variables, taper du code php executé dans le contexte de breakpoint, bref déboger efficacement.

### Config de port docker

Il n'y a pas de configuration à faire si vous utiliser l'adresse de votre machine vue par docker. Usuellement quelque chose qui ressemble à 172.18.0.1.  
Par exemple pour le contenaire de dev de ce blog, je lance `docker inspect blog_hugo | grep Gateway` et j'ai l'ip de ma machine sur le réseau docker.

## Les clés à emporter

Pour lancer un script php à déboger quand l'ide ne peut pas le faire :

1. Définir un serveur php sur votre phpstorm en utilisant le _path mapping_ et noter son nom.
2. Ajouter des variables d'environnement dans la commande shell qui lance votre script : `XDEBUG_MODE=debug XDEBUG_SESSION=1 XDEBUG_CONFIG="client_host=172.17.0.1 client_port=9003" PHP_IDE_CONFIG="serverName=lePetitNomDeMonServerDansPhpstorm" php /app/hello.php`

## Pour aller plus loin

Je continue de dockerisé complètement mon environnement de travail. Il faudra voir pour ce qui concerne les pages web maintenant, ça viendra quand j'en aurais besoin.  
A ce moment, je serais peut-être amené à revoir ma config.




src/config.ts
import type { Site, SocialObjects } from "./types";

export const SITE: Site = {
  website: "https://blog.seb7.fr/", // replace this with your deployed domain
  author: "Sebastien Monterisi",
  profile: "https://github.com/sebsept/",
  desc: "Quelques notes, impressions, avis, réalisations, apprentissages, d'un développeur web.",
  title: "DevBlog - Sébastien Monterisi",
  ogImage: "astropaper-og.jpg",
  lightAndDarkMode: false,
  postPerIndex: 7,
  postPerPage: 7,
  scheduledPostMargin: 15 * 60 * 1000, // 15 minutes
  showArchives: false,
  editPost: {
    url: "https://github.com/SebSept/blog/blob/main/src/content/blog",
    text: "Suggestion",
    appendFilePath: true,
  },
  minTagOccurrences: 2,
};

export const LOCALE = {
  lang: "fr", // html lang code. Set this empty and default will be "en"
  langTag: ["fr-FR"], // BCP 47 Language Tags. Set this empty [] to use the environment default
} as const;

export const LOGO_IMAGE = {
  enable: false,
  svg: true,
  width: 216,
  height: 46,
};

export const SOCIALS: SocialObjects = [
  {
    name: "LinkedIn",
    href: "https://www.linkedin.com/in/sebastienmonterisi/",
    linkTitle: `Mon LinkedIn`,
    active: true,
  },
  {
    name: "Github",
    href: "https://github.com/SebSept",
    linkTitle: `Mon Github`,
    active: true,
  },
  {
    name: "Mastodon",
    href: "https://phpc.social/@sebastienmonterisi",
    linkTitle: `${SITE.title} on Mastodon`,
    active: true,
  },
  {
    name: "Mail",
    href: "mailto:sebastienmonterisi@gmail.com",
    linkTitle: `Envoyez moi un mail`,
    active: true,
  },
  {
    name: "X",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on X`,
    active: false,
  },
  {
    name: "Twitch",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Twitch`,
    active: false,
  },
  {
    name: "YouTube",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on YouTube`,
    active: false,
  },
  {
    name: "WhatsApp",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on WhatsApp`,
    active: false,
  },
  {
    name: "Snapchat",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Snapchat`,
    active: false,
  },
  {
    name: "Pinterest",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Pinterest`,
    active: false,
  },
  {
    name: "TikTok",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on TikTok`,
    active: false,
  },
  {
    name: "CodePen",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on CodePen`,
    active: false,
  },
  {
    name: "Discord",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Discord`,
    active: false,
  },
  {
    name: "GitLab",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on GitLab`,
    active: false,
  },
  {
    name: "Reddit",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Reddit`,
    active: false,
  },
  {
    name: "Skype",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Skype`,
    active: false,
  },
  {
    name: "Steam",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Steam`,
    active: false,
  },
  {
    name: "Telegram",
    href: "https://github.com/satnaing/astro-paper",
    linkTitle: `${SITE.title} on Telegram`,
    active: false,
  },
];
